### PR TITLE
Add Node model and its queries

### LINKED ISSUE #4: User Story #4 - Creating Node Objects in Parse and Java Classes
- Creating node objects in parse dashboard
- Outlining Java classes

### PR DESCRIPTION
**Pull request by LeoNero**
Closes #4

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Looks good

File: app/src/main/java/com/fbu/fbuteam/models/Node.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        put(KEY_TYPE, type);
+    }
+
+    public Node getParent() {
+        ParseObject parent = getParseObject(KEY_PARENT);
+
+        if (parent != null) {
+            return (Node) parent;
+        }
+
```
Comment (haleyharelson) [inline app/src/main/java/com/fbu/fbuteam/models/Node.java:47]: maybe delete empty line here to make spacing more consistent

File: app/src/main/java/com/fbu/fbuteam/models/Node.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        }
+    }
+
+    public List<Node> getChildren() {
+        Object children = get("children");
+
+        if (children != null) {
+            return (List<Node>) children;
+        }
+
```
Comment (haleyharelson) [inline app/src/main/java/com/fbu/fbuteam/models/Node.java:63]: ^ see comment above

File: app/src/main/java/com/fbu/fbuteam/models/Node.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        if (parent != null) {
+            node.setParent(parent);
+        }
+
+        return node;
+    }
+
+    private static Node createDefaultNode(String name, int level, String type) {
```
Comment (haleyharelson) [inline app/src/main/java/com/fbu/fbuteam/models/Node.java:97]: why do we need to create a default node?

**Review by haleyharelson — APPROVED** [review]

**Pull request merged**